x-pack/plugins/ml/public/application/jobs/new_job/common/map_loader/map_loader.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import memoizeOne from 'memoize-one';
+import { isEqual } from 'lodash';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import { ES_GEO_FIELD_TYPE, LayerDescriptor } from '@kbn/maps-plugin/common';
+import type { MapsStartApi } from '@kbn/maps-plugin/public';
+import { ChartLoader } from '../chart_loader';
+import { Field, SplitField } from '../../../../../../common/types/fields';
+const eq = (newArgs: any[], lastArgs: any[]) => isEqual(newArgs, lastArgs);
+
+export class MapLoader extends ChartLoader {
+  private _indexPatternId: string | undefined;
```

Comment:
I think we should changes the base class to store the whole "indexPattern" naming it the `_dataView` and remove the `_indexPatternTitle` and `_indexPatternId` member variables.

Reply:
Good call- updated in 9be34fed1d464e8fac17090c2aa5e2e78099e150